### PR TITLE
Assignment3A, Assignment3B and Assignment3C

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**

### CODE REVIEW

File: SCF/Assignment3A/StringManipulation.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		 */
+		for (int charIndex = stringToBeReversed.length() - 1; charIndex >= 0; charIndex--) 
+		{
+			reversedString.append(stringToBeReversed.charAt(charIndex));
+		}
+		
+		/**
+		 * making @reversedString immutable.
+		 */
+		String finalreversedString = reversedString.toString();
```
Comment (rash-tiwari) [inline SCF/Assignment3A/StringManipulation.java]: merge the line ```80``` and ```81```  to ```return reversedString.toString();``` . Do the same for other methods as well.

File: SCF/Assignment3A/TestStringManipulation.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+public class TestStringManipulation 
+{
+
+	public static void main(String[] args) 
+	{	
+		StringManipulation stringManipulation = new StringManipulation();
+		Scanner userInput = new Scanner(System.in);
+
+		//while loop for options
+		while (true)
```
Comment (rash-tiwari) [inline SCF/Assignment3A/TestStringManipulation.java]: Use ```Do-While``` instead of ```while```

File: SCF/Assignment3B/Marksheet.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	}
+	
+	/**
+	 * 
+	 * Find percentage of student passed
+	 * @param grades : array containing grades of all student
+	 * @param numberOfStudent : total number of students.
+	 * @return percentage of student passed
+	 */
+	public double getPercentageOfPassStudent(double grades[],int numberOfStudent) throws IOException
```
Comment (rash-tiwari) [inline SCF/Assignment3B/Marksheet.java]: Add space after comma ```,```  in parameter ```double grades[],int numberOfStudent````

File: SCF/Assignment3B/TestMarksheet.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		
+		//taking student grades
+		for(int studentIndex = 0; studentIndex < numberOfStudent; studentIndex++)
+		{
+			System.out.print("Enter Grade Of "+ (studentIndex+1)+" Student : ");
+			studentGrades[studentIndex] = userInput.nextDouble();
+		}
+		
+		//option selecting loop
+		while(true)
```
Comment (rash-tiwari) [inline SCF/Assignment3B/TestMarksheet.java]: Use ```do-while``` instead of ```while```

File: SCF/Assignment3B/TestMarksheet.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+public class TestMarksheet {
+
+	public static void main(String[] args)
+	{
+		Marksheet marksheet = new Marksheet();
+		Scanner userInput = new Scanner(System.in);
+		
+		//taking number of student input
+		System.out.println("Enter Number Of Student : ");
+		int numberOfStudent = userInput.nextInt();
```
Comment (rash-tiwari) [inline SCF/Assignment3B/TestMarksheet.java:21]: ``` Add validation message```  if numberOfstudent is less than or equal to zero

File: SCF/Assignment3B/TestMarksheet.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+					try
+					{
+						double maximumGradesOfAllStudent = marksheet.getMaximumOfAllGrades(studentGrades, numberOfStudent);
+						System.out.printf("Maximum Grades Of All Student is : %.2f \n", maximumGradesOfAllStudent);
+					}
+					catch(IOException io)
+					{
+						System.out.println(io);
+					}
+
```
Comment (rash-tiwari) [inline SCF/Assignment3B/TestMarksheet.java]: Remove unnecessary line from here and from similar place in file

File: SCF/Assignment3C/TestArea.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+public class TestArea 
+{
+
+	public static void main(String[] args) 
+	{
+		
+		Area area = new Area();
+		Scanner userInput = new Scanner(System.in);
+		
+		while(true)
```
Comment (rash-tiwari) [inline SCF/Assignment3C/TestArea.java]: Get change while to do while

File: SCF/Assignment3C/TestArea.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+						System.out.print("Enter Base Of Right Angle Triangle : ");
+						double baseOfRightAngleTriangle = userInput.nextDouble();
+						
+						double areaOfRightAngleTriangle = area.getAreaOfTriangle(heightOfRightAngleTriangle, baseOfRightAngleTriangle);
+						System.out.printf("Area Of Right Angle Triangle is : %.2f \n", areaOfRightAngleTriangle);
+						
+					}
+					catch(Exception e)
+					{
+						System.out.println(e);
```
Comment (rash-tiwari) [inline SCF/Assignment3C/TestArea.java]: Use ```e.printStackTrace()``` at every place for printing exception message.

**Review by rash-tiwari — CHANGES_REQUESTED** [review]
Please do the changes.

File: SCF/Assignment3A/StringManipulation.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+ * This class designed for provide methods of string manipulation.
+ * This class contain stringComparator, getReverse, replaceLowerCaseWithUpperCase and getLargestWord methods.
+ */
+public class StringManipulation 
+{	
+	/**
+	 * 
+	 * @param firstString : first string which we want to compare.
+	 * @param secondString : second string which we want to compare.
+	 * @fisrtString or @secondString should not be null.
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Assignment3A/StringManipulation.java]: Done

File: SCF/Assignment3A/StringManipulation.java (hunk 1 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3A/StringManipulation.java]: Done

File: SCF/Assignment3A/TestStringManipulation.java (hunk 2 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3A/TestStringManipulation.java]: Done

File: SCF/Assignment3B/Marksheet.java (hunk 3 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3B/Marksheet.java]: Done

File: SCF/Assignment3B/TestMarksheet.java (hunk 5 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3B/TestMarksheet.java:21]: Done

File: SCF/Assignment3B/TestMarksheet.java (hunk 4 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3B/TestMarksheet.java]: Done

File: SCF/Assignment3B/TestMarksheet.java (hunk 6 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3B/TestMarksheet.java]: Done

File: SCF/Assignment3C/TestArea.java (hunk 8 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3C/TestArea.java]: Done

File: SCF/Assignment3C/TestArea.java (hunk 7 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment3C/TestArea.java]: Done

**Review by rash-tiwari — COMMENTED** [review]
approved

**Review by rash-tiwari — APPROVED** [review]

**Pull request merged**